pkgs/by-name/re/refine/package.nix

```
@@ -0,0 +1,79 @@
+{
+  lib,
+  fetchFromGitLab,
+  nix-update-script,
+  blueprint-compiler,
+  desktop-file-utils,
+  libadwaita,
+  meson,
+  ninja,
+  pkg-config,
+  wrapGAppsHook4,
+  python3,
+  libxml2,
+  python3Packages,
+  libportal,
+  libportal-gtk4,
+  appstream,
+  glib,
+  gtk4,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "refine";
+  version = "0.2.0";
+  pyproject = false; # uses meson
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "TheEvilSkeleton";
+    repo = "Refine";
+    tag = version;
+    hash = "sha256-NmfsogXkJicZj/cE9HFz91ERD9fxOGKoBV4h68KIE+g=";
+  };
+
+  nativeBuildInputs = [
+    appstream
+    blueprint-compiler
+    glib # For `glib-compile-schemas`
+    gtk4 # For `gtk-update-icon-cache`
+    meson
+    ninja
+    pkg-config
+    wrapGAppsHook4
+    python3
+    desktop-file-utils
+  ];
+
+  strictDeps = true;
+
+  buildInputs = [
+    libxml2
+    libadwaita
+  ];
+
+  dependencies =
+    [
+      libportal
+      libportal-gtk4
+    ]
+    ++ (with python3Packages; [
+      pygobject3
+    ]);
+
+  mesonFlags = [ (lib.mesonBool "network_tests" false) ];
+
```

```suggestion
  dontWrapGApps = true;

  makeWrapperArgs = [ "\${gappsWrapperArgs[@]}" ];
  
  # NOTE: `postCheck` is intentionally not used here, as the entire checkPhase
  # is skipped by `buildPythonApplication`
  # https://github.com/NixOS/nixpkgs/blob/9d4343b7b27a3e6f08fc22ead568233ff24bbbde/pkgs/development/interpreters/python/mk-python-derivation.nix#L296
  postInstallCheck = ''
    mesonCheckPhase
  '';
```
Just realized we're being double wrapped. We can avoid that by adding the GTK wrapper's args to the ones `buildPythonPackage` uses

Lets try to run the tests from Meson we have the flag for as well :p